plugins/deployed-labeler/main.go

```diff
@@ -90,27 +90,44 @@ func main() {
 }
 
 func (s *server) markDeployedPR(w http.ResponseWriter, req *http.Request) {
-	switch req.Method {
-	case "POST":
-		var commitSHA string
-		var team string
-		for k, v := range req.URL.Query() {
-			switch k {
-			case "commit":
-				commitSHA = v[0]
-			case "team":
-				team = v[0]
-			default:
-				s.log.Warnf("Unrecognized parameter received: %s", k)
-			}
+	var commitSHA string
+	var team string
+	for k, v := range req.URL.Query() {
+		switch k {
+		case "commit":
+			commitSHA = v[0]
+		case "team":
+			team = v[0]
+		default:
+			s.log.Warnf("Unrecognized parameter received: %s", k)
 		}
-		if team == "" || commitSHA == "" {
-			w.WriteHeader(http.StatusPreconditionFailed)
-			w.Write([]byte(http.StatusText(http.StatusPreconditionFailed)))
-			s.log.Errorf("team and commit are required parameters. Team: %v, commit: %v", team, commitSHA)
-			break
+	}
+	if team == "" || commitSHA == "" {
+
+		var preconditionFailed struct {
+			Errors []string
 		}
 
+		preconditionFailed.Errors = make([]string, 0, 1)
+		preconditionFailed.Errors = append(preconditionFailed.Errors, fmt.Sprintf("team and commit are required parameters. Team: '%v', commit: '%v'", team, commitSHA))
```

Comment:
What was your goal in creating a new type to store the error?

Could we use standard error libs? Maybe [errors](https://pkg.go.dev/errors) would make more sense here

Reply:
> What was your goal in creating a new type to store the error?

The PR mention's _per https://github.com/golang/go/issues/5161 serializing an error object to json emits an empty object; this caused deployed-labeler to emit an empty object when erroring. This commit converts all errors to strings before returning to the user to provide more context._

Seems to be serialisation and making sure the errors are returned in the HTTP response. 🤔

Reply:
Update! New err type is not required. Rolled it back to `error`